debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java

```diff
@@ -85,64 +92,51 @@ private PostgresReplicationConnection(Configuration config,
         this.plugin = plugin;
         this.dropSlotOnClose = dropSlotOnClose;
         this.statusUpdateInterval = statusUpdateInterval;
+        this.exportSnapshot = exportSnapshot;
         this.messageDecoder = plugin.messageDecoder();
         this.typeRegistry = typeRegistry;
         this.streamParams = streamParams;
+        this.slotCreationInfo = Optional.empty();
+        this.hasInitedSlot = false;
 
+        // to keep the same
         try {
-            initReplicationSlot();
-        }
-        catch (ConnectException e) {
-            close();
-            throw e;
-        }
-        catch (Throwable t) {
+            ensureSlotNotActive(getSlotInfo());
```

Comment:
This is probably the most tricky change in the whole commit, rather than always connecting right away, we now delay connecting which allows us to have more control over the lifecycle of when a snapshot is created, specifically, this needed to happen so that when you a export a snapshot, no additional queries are done on the slot until the snapshot is done, otherwise, that invalidates the exported snapshot.

Reply:
Gotcha; thanks.